source/served/ddoc.d

```diff
@@ -3,14 +3,101 @@ module served.ddoc;
 import served.protocol;
 
 import std.string;
+import std.format;
 import ddoc;
 
+
+/**
+ * A test function for checking `DDoc` parsing
+ *
+ * Params:
+ *     hello = a string
+ *     world = an integer
+ *
+ * Author: Jonny
+ * Bugs: None
+ * ---
+ * import std.stdio;
+ *
+ * int main(string[] args) {
+ * 	   writeln("Testing inline code")
+ * }
+ * ---
+ */
+private int testFunction(string foo, int bar)
+{
+	import std.stdio : writeln;
+	writeln(foo, bar);
+	return 0;
+}
+
+
+/**
+ * Convert a Ddoc comment string to markdown. Returns ddoc string back if it is
+ * not valid.
+ * Params:
+ *		ddoc = string of a valid Ddoc comment.
+ */
 string ddocToMarkdown(string ddoc)
 {
-	auto lexer = Lexer(prepareDDoc(ddoc), true);
-	return expand(lexer, markdownMacros, false).replace("&#36;", "$");
+	// Parse ddoc. Return if exception.
+	Comment comment;
+	try
+	{
+		comment = parseComment(prepareDDoc(ddoc), markdownMacros);
+	}
+	catch (Exception e)
+	{
+		return ddoc;
+	}
+
+	auto output = "";
+	foreach(Section section; comment.sections)
+	{
+		import std.uni : toLower;
+		switch(section.name.toLower)
+		{
+		case "":
+		case "summary":
+			output ~= section.content ~= "\n\n";
+			break;
+		case "params":
+			output ~= "**Params**\n\n";
+			foreach (parameter; section.mapping)
+			{
+				output ~= format!"`%s` %s\n\n"(parameter[0], parameter[1]);
+			}
+			break;
+		case "author":
+		case "authors":
+		case "bugs":
+		case "date":
+		case "deprecated":
+		case "history":
+		default:
+			// Single line sections go on the same line as section titles. Multi
+			// line sections go on the line below.
+			import std.algorithm : find;
+			if (section.content.chomp().find('\n').empty)
```

Comment:
I think canFind() would be cleaner here and you could extend this to `canFind(only("\r\n", "\n", "\r"))`

Reply:
Right. I'm not super familiar with all the of `std.algorithm` yet.